Golem.Tests/JobTests.cs

```diff
@@ -95,10 +95,10 @@ public async Task CompleteScenario()
             _logger.LogInformation("Starting Golem");
             await golem.Start();
             // On startup Golem status goes from `Off` to `Starting`
-            Assert.Equal(GolemStatus.Starting, await SkipMatching(golemStatusChannel, (GolemStatus s) => s == GolemStatus.Off));
+            Assert.Equal(GolemStatus.Starting, await ReadChannel(golemStatusChannel, (GolemStatus s) => s == GolemStatus.Off));
```

Comment:
Name `SkipMatching` was confusing. `matcher` arg has a code comment indicating what it is for.